KalturaWowzaServer/src/main/java/com/kaltura/media_server/services/KalturaEntryDataPersistence.java

```diff
@@ -0,0 +1,114 @@
+                Set<String> hashedEntriesList = entryIdToKalturaLiveEntryMap.keySet();
+                long currentTime = System.currentTimeMillis();
+                for (String entry : hashedEntriesList) {
+                    // Check start time to avoid a race between the thread adding the entry to the map and this
+                    // current thread that wants to remove it. Worst case scenario entry will ve erased in the next run.
+                    long startTime = (long)getPropertyByEntry(entry, Constants.KALTURA_ENTRY_START_TIME);
+                    if (!playingEntriesList.contains(entry)) {
+                        logger.info("(" + entry + ") Entry no longer exists");
+                        int minTimeInMap = Constants.KALTURA_PERSISTENCE_DATA_MIN_ENTRY_TIME;
+                        if (currentTime - startTime > minTimeInMap) {
+                            logger.info("(" + entry + ") Entry start time is greater than " + minTimeInMap / 1000 + " seconds; Removing it from Map!");
+                            entryIdToKalturaLiveEntryMap.remove(entry);
+                        }
+                    }
+                }
+            }
+        }
+        catch (Exception e) {
+            logger.error("Error occurred while cleaning Persistence Data map: " + e);
+        }
+	}
+
+	public static Object getPropertyByStream(String streamName, String subKey) throws Exception {
+        try {
+            String entryIdKey = Utils.getEntryIdFromStreamName(streamName);
+            return getPropertyByEntry(entryIdKey, subKey);
+        }
+        catch (Exception e) {
+            logger.error("(" + streamName + ") Stream failed to retrieve value form entry. Error: " + e);
+            throw e;
+        }
+	}
+
+	public static Object getPropertyByEntry(String entryId, String subKey) throws Exception {
+
+		Object value;
+
+		try {
+			ConcurrentHashMap<String, Object> entryMap = entryIdToKalturaLiveEntryMap.get(entryId);
+			value = entryMap.get(subKey);
+		} catch (Exception e) {
+			logger.error("(" + entryId + ") Failed to get value for key \"" + subKey + "\". " + e);
+			throw e;
+		}
+
+		logger.debug("(" + entryId + ") Successfully got entry subKey: (" + subKey + ")");
+
+		return value;
+	}
+
+	// note: call add entry on connect (after successful authentication)
+	public static void setProperty(String entryId, String subKey, Object value) throws Exception {
+		synchronized (entryIdToKalturaLiveEntryMap) {
+			entryIdToKalturaLiveEntryMap.putIfAbsent(entryId, new ConcurrentHashMap<String, Object>());
+			setValueProperty(entryId, subKey, value);
+		}
+	}
+
+	private static void setValueProperty(String entryId, String subKey, Object value) throws Exception {
+		ConcurrentHashMap<String, Object> entryMap = entryIdToKalturaLiveEntryMap.get(entryId);
+		entryMap.put(subKey, value);
+		long time = System.currentTimeMillis();
+		entryMap.put(Constants.KALTURA_ENTRY_START_TIME, time);
```

Comment:
rename Constants.KALTURA_ENTRY_START_TIME to Constants.KALTURA_ENTRY_VALIDATED_TIME